src/gui/annotator/AnnotationWidget.cpp

```diff
@@ -59,6 +59,11 @@ void AnnotationWidget::initGui()
 	insertDockWidget(Qt::TopDockWidgetArea, mItemSettings);
 	insertDockWidget(Qt::BottomDockWidgetArea, mGeneralSettings);
 	insertDockWidget(Qt::BottomDockWidgetArea, mImageSettings);
+	insertDockWidget(Qt::BottomDockWidgetArea, mControlsWidget);
+
+	// hide the last inserted DockWidget (mControlsWidget) since we want to show it only if needed using setControlsWidgetVisible(true)
+	mControlsDockWidget = mDockWidgets.last();
+	removeDockWidget(mControlsDockWidget);
```

Comment:
This is a bit strange code, you add something (which might not be used at all) then remove it right away. I would recommend first to change the insertDockWidget to return the inserted DockWidget so you don't have to relay on the `.last()`  call. Then I would remove this adding here and extend the `setControlsWidgetVisible` method to check if the `mControlsDockWidget` is `nullptr`, yes then insert a new dock widget and assign it to `mControlsDockWidget`. If it's not longer nullptr the do what you're doing already in the `setControlsWidgetVisible` method.

Reply:
Unfortunately [addDockWidget (QT docs)](https://doc.qt.io/qt-5/qmainwindow.html#addDockWidget) doesn't return anything. Maybe return `.last()` from within `insertDockWidget()` and then check if `mControlsWidget` is `nullptr`?

Reply:
Yes, but that is our method, you can change `insertDockWidget` to return the created object, the one that is called `dockWidget` in this method.

Reply:
Like this? (check latest commit)

Reply:
Yes, exactly.